### PR TITLE
Civil Servant Services view

### PR DESCRIPTION
**Pull request by TheFehr**

### CODE REVIEW

File: spec/models/civil_servant_spec.rb [hunk 1]
```diff
@@ -86,7 +86,7 @@
   end
 
   describe '#active?' do
```
Comment (lukasbischof) [inline spec/models/civil_servant_spec.rb]: Renamed

File: app/views/civil_servants/overview/index.html.slim [hunk 2]
```diff
@@ -1,27 +1,39 @@
...
+        text: 'Stiftung SWO',
+        type: OverviewCardComponent::TYPES[:info],
+        links: { t('.links.all_services') => civil_servants_services_path } \
+      ) \
+    )
+  - else
+    = render(\
+          OverviewCardComponent.new(\
+            title: '',
+            text: 'Kein aktueller Einsatz',
```
Comment (lukasbischof) [inline app/views/civil_servants/overview/index.html.slim:15]: Can you please translate that?

File: app/views/civil_servants/overview/index.html.slim [hunk 3]
```diff
@@ -1,27 +1,39 @@
...
-      title: 'Momentaner Einsatz',
-      text: 'Stiftung SWO',
-      type: OverviewCardComponent::TYPES[:warning],
-      links: { 'See all' => '#' } \
-    ) \
-  )
+  - if @civil_servant.in_service?
+    = render( \
+      OverviewCardComponent.new( \
+        title: 'Momentaner Einsatz',
```
Comment (lukasbischof) [inline app/views/civil_servants/overview/index.html.slim:5]: Can you please translate that?

File: app/views/civil_servants/overview/index.html.slim [hunk 4]
```diff
@@ -1,27 +1,39 @@
...
+        text: 'Stiftung SWO',
```
Comment (lukasbischof) [inline app/views/civil_servants/overview/index.html.slim:6]: Can you please make that dynamic

File: app/views/civil_servants/overview/index.html.slim [hunk 5]
```diff
@@ -1,27 +1,39 @@
...
-      title: 'Einsatz bis',
-      text: '20. Jun 2020',
-      type: OverviewCardComponent::TYPES[:success],
-      links: { 'Zum Einsatz' => '#' } \
-    ) \
-  )
+  - if current_civil_servant.in_service?
+    = render( \
+      OverviewCardComponent.new( \
+        title: 'Einsatz bis',
```
Comment (lukasbischof) [inline app/views/civil_servants/overview/index.html.slim:33]: Can you please translate that?

File: app/views/civil_servants/services/_expense_sheet_short_info_row.html.slim [hunk 6]
```diff
@@ -0,0 +1,5 @@
+tr
+  td = l(expense_sheet.beginning)
+  td = l(expense_sheet.ending)
+  td = expense_sheet.duration
+  td = link_to('#') { t('.print') }
```
Comment (lukasbischof) [inline app/views/civil_servants/services/_expense_sheet_short_info_row.html.slim:5]:
```suggestion
  td = link_to(t('.print'), '#')
```

File: app/views/civil_servants/services/show.html.slim [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    table.table
+      thead.thead-light
+        tr
+          th scope="col" = t('activerecord.attributes.expense_sheet.beginning')
+          th scope="col" = t('activerecord.attributes.expense_sheet.ending')
+          th scope="col" = t('.duration')
+          th scope="col"
+      tbody = render partial: 'civil_servants/services/expense_sheet_short_info_row',
+              collection: @service.expense_sheets, as: :expense_sheet
+
```
Comment (lukasbischof) [inline app/views/civil_servants/services/show.html.slim:31]:
```suggestion
              collection: @service.expense_sheets, as: :expense_sheet
```

File: spec/models/service_spec.rb [hunk 8]
```diff
@@ -211,8 +213,10 @@
     end
   end
 
-  describe '#in_future?' do
-    subject { build(:service, :last, beginning: beginning).in_future? }
+  describe '#future?' do
+    subject { build(:service, :last, beginning: beginning, ending: ending).future? }
+
+    let(:ending) { beginning + 26.days }
```
Comment (lukasbischof) [inline spec/models/service_spec.rb:219]: Isn't it 25 dayzz

```suggestion
    let(:ending) { beginning + 25.days }
```

File: spec/requests/civil_servants/services_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+  describe '#show' do
+    let(:perform_request) { get civil_servants_service_path(service) }
+
+    let(:service) { create :service }
+
+    context 'when a civil servant is signed in' do
+      let(:civil_servant) { service.civil_servant }
+
+      before { sign_in civil_servant.user }
```
Comment (lukasbischof) [inline spec/requests/civil_servants/services_controller_spec.rb:125]: You don't use the let but in the before

```suggestion
      before { sign_in service.civil_servant.user }
```

File: spec/requests/civil_servants/services_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+      let(:organization_admin) { create :organization_member }
+
+      before do
+        sign_in organization_admin.user
+        perform_request
+      end
+
+      it_behaves_like 'unauthorized request'
+    end
+
```
Comment (lukasbischof) [inline spec/requests/civil_servants/services_controller_spec.rb:143]: Probably, also a `context 'when a civil servants tries to access a service from a different civil servant'`?

File: spec/views/civil_servants/services/show.html.slim_spec.rb [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  let(:expected_strings) do
+    [
+      t('civil_servants.services.show.organization_information'),
+      organization.name
+    ] + [
+      org_address.primary_line,
+      org_address.secondary_line,
+      org_address.supplement,
+      org_address.street,
+      org_address.zip_with_city
```
Comment (lukasbischof) [inline spec/views/civil_servants/services/show.html.slim_spec.rb:26]: Isn't this just `org_address.full_compose`?

Comment (TheFehr) [reply spec/views/civil_servants/services/show.html.slim_spec.rb:26]: No because full compose is one string. But on the view the \n are replaced with <br>

**Pull request merged**